filebeat/prospector/redis/harvester.go

```diff
@@ -0,0 +1,108 @@
+package redis
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	rd "github.com/garyburd/redigo/redis"
+)
+
+type Harvester struct {
+	outlet channel.Outleter
+}
+
+// log contains all data related to one slowlog entry
+//
+// 	The data is in the following format:
+// 	1) (integer) 13
+// 	2) (integer) 1309448128
+// 	3) (integer) 30
+// 	4) 1) "slowlog"
+// 	   2) "get"
+// 	   3) "100"
+//
+type log struct {
+	id        int64
+	timestamp int64
+	duration  int
+	cmd       string
+	key       string
+	args      []string
+}
+
+func (h *Harvester) Start(conn rd.Conn) error {
+	defer conn.Close()
+
+	logp.Debug("redis", "Start slowlog fetch.")
+
+	// Writes Slowlog get and slowlog reset both to the buffer so they are executed together
+	conn.Send("SLOWLOG", "GET")
+	conn.Send("SLOWLOG", "RESET")
+
+	// Flush the buffer to execute both commands and receive the reply from SLOWLOG GET
+	conn.Flush()
+
+	// Receives first reply from redis which is the one from GET
+	logs, err := rd.Values(conn.Receive())
+	if err != nil {
+		return fmt.Errorf("error receiving slowlog data: %s", err)
+	}
+
+	// Read reply from RESET
+	_, err = conn.Receive()
+	if err != nil {
+		return fmt.Errorf("error receiving reset data: %s", err)
+	}
+
+	for _, item := range logs {
+		entry, err := rd.Values(item, nil)
+		if err != nil {
+			logp.Err("Error loading slowlog values: %s", err)
+			continue
+		}
+
+		var log log
+		var args []string
+		rd.Scan(entry, &log.id, &log.timestamp, &log.duration, &args)
+
+		// This splits up the args into cmd, key, args. Should there also be an entry with
+		// all in one line.
+		argsLen := len(args)
+		if argsLen >= 1 {
+			log.cmd = args[0]
+		}
+		if argsLen >= 2 {
+			log.key = args[1]
+		}
+		// TODO: Could args contain confidential data? Should it be optional?
+		if argsLen >= 3 {
+			log.args = args[2:]
+		}
+
+		data := util.NewData()
+		data.Event = common.MapStr{
+			// TODO: Should we still send the "fetching" timestamp or overwrite it by timestamp?
```

Comment:
I decided for option 2 for the moment. The reasons is that I think it most cases that is what people expect in `@timestamp` here. So also without postprocessing in ingest, the data is already correct.